account_banking_reconciliation/account_banking_reconciliation.py

```diff
@@ -0,0 +1,1224 @@
+                 - sres['cleared_balance_in_currency']
+                 - statement.starting_balance_in_currency),
+                account_precision)
+            sres['general_ledger_balance'] = self._get_gl_balance(
+                cr, uid, statement.account_id.id, statement.ending_date)
+            sres['registered_balance'] = round(
+                (statement.starting_balance
+                 + sres['cleared_balance']
+                 + sres['uncleared_balance']),
+                account_precision)
+
+        return res
+
+    def _get_gl_balance(self, cr, uid, account_id, date=None):
+        """ Get the General Ledger balance at date for account """
+        fyear_obj = self.pool["account.fiscalyear"]
+        account_obj = self.pool['account.account']
+        date = date or time.strftime('%Y-%m-%d')
+        fiscal_yearid = fyear_obj.find(cr, uid, date)
+        date_from = fyear_obj.browse(cr, uid, fiscal_yearid).date_start
+
+        balance = account_obj.read(
+            cr, uid, account_id, ['balance'],
+            context={'date_from': date_from,
+                     'date_to': date,
+                     'state': 'posted'})
+        return balance["balance"]
+
+    def _get_move_line_write(self, line):
+        res = {
+            'ref': line.ref,
+            'date': line.date,
+            'partner_id': line.partner_id.id,
+            'currency_id': line.currency_id.id,
+            'amount': line.credit or line.debit,
+            'name': line.name,
+            'move_line_id': line.id,
+            'type': line.credit and 'cr' or 'dr'
+        }
+
+        if line.credit:
+            res['amount_in_currency'] = -line.amount_currency
+        else:
+            res['amount_in_currency'] = line.amount_currency
+
+        return res
+
+    def refresh_record(self, cr, uid, ids, context=None):
+        account_move_line_obj = self.pool["account.move.line"]
+        for obj in self.browse(cr, uid, ids, context=context):
+            if not obj.account_id:
+                continue
+
+            to_write = {'credit_move_line_ids': [], 'debit_move_line_ids': []}
+            # For existing records, allow refresh to set multi_currency
+            # correctly by updating it during the refresh. This allows viewing
+            # multi currency fields without changing the account
+            acur = obj.account_id.currency_id
+            ccur = obj.account_id.company_id.currency_id
+            if acur and ccur and acur.id != ccur.id:
+                to_write['multi_currency'] = True
+
+            move_line_ids = [
+                line.move_line_id.id
+                for line in obj.credit_move_line_ids + obj.debit_move_line_ids
+                if line.move_line_id
+            ]
+
+            domain = [
+                ('id', 'not in', move_line_ids),
+                ('account_id', '=', obj.account_id.id),
+                ('move_id.state', '=', 'posted'),
+                ('cleared_bank_account', '=', False),
```

Comment:
as @fanzalika pointed out, `('journal_id.type', '!=', 'situation')` should be added